data/org.freedesktop.portal.GlobalShortcuts.xml

```diff
@@ -0,0 +1,184 @@
+      <arg type="o" name="handle" direction="out"/>
+      <annotation name="org.qtproject.QtDBus.QtTypeName.In0" value="QVariantMap"/>
+    </method>
+
+      <!--
+        ListShortcuts:
+        @session_handle: Object path for the #org.freedesktop.impl.portal.Session object representing the session
+        @options: Unused so far
+
+        List all the shortcuts with their invocation descriptions.
+
+        The following results get returned via the #org.freedesktop.portal.Request::Response signal:
+        <variablelist>
+          <varlistentry>
+            <term>shortcuts a(sa{sv})</term>
+            <listitem><para>
+              The different shortcut ids that have been registered. They are mapped to a list of pairs including:
+                - description: User-readable text describing what the shortcut does
+                - trigger: User-readable text describing how to trigger the shortcut
+            </para></listitem>
+          </varlistentry>
+        </variablelist>
+    -->
+    <method name="ListShortcuts">
+      <arg type="o" name="session_handle" direction="in"/>
+      <arg type="o" name="request_handle" direction="out"/>
+    </method>
+
+    <!--
+        BindShortcuts:
+        @session_handle: Object path for the #org.freedesktop.impl.portal.Session object representing the session
+        @shortcuts: The identifier of the shortcuts we intend to register, empty for all shortcuts
+        @request_handle: Object path for the #org.freedesktop.portal.Request object representing this call
+
+        Shows a dialog with all shortcuts to configure and offers them back to the app when done.
+
+        The following results get returned via the #org.freedesktop.portal.Request::Response signal:
+        <variablelist>
+          <varlistentry>
+            <term>shortcuts a(sa{sv})</term>
+            <listitem><para>
+              The different shortcut ids that have been registered. They are mapped to a list of pairs including:
+                - description: User-readable text describing what the shortcut does
+                - trigger: User-readable text describing how to trigger the shortcut
+            </para></listitem>
+          </varlistentry>
+        </variablelist>
+    -->
+    <method name="BindShortcuts">
+      <arg type="o" name="session_handle" direction="in"/>
+      <arg type="as" name="shortcuts" direction="in"/>
+      <arg type="s" name="parent_window" direction="in"/>
+      <arg type="a{sv}" name="options" direction="in"/>
+      <arg type="o" name="request_handle" direction="out"/>
+
+      <annotation name="org.qtproject.QtDBus.QtTypeName.In3" value="QVariantMap"/>
+    </method>
+
+    <!--
+        Activated:
+        Notifies about a shortcut becoming active.
+
+        @session_handle: Session that requested the shortcut
+        @shortcut_id: the application-provided ID for the notification
+        @timestamp: When the event happened in ms
```

Comment:
Does it make any sense for timestamps to have a unit like this? I always thought generic timestamps like this one were just opaque numbers that don't really represent any specific time or range, and should be used strictly for comparison.

Reply:
I've seen it backfire. When I was implementing the recording of pipewire streams into ffmpeg I found the lack of units a big problem when reading documentation. I understand it's not the same case though, still I think it's good to say what we are talking about.

We are not providing a frame of reference, so it's still for comparison anyway.

Reply:
Specifying unit seems good, but maybe not the origin (e.g. what clock other than it's not something affected by timezone changes or real time clock changes). Maybe make it microseconds instead though? It's the source of most timestamps; milliseconds feels more like an X11 relic.

Reply:
Changing to µs